lib/practice_exercises.rb

```diff
@@ -1,13 +1,50 @@
 
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: 0(n)
+# Space Complexity: 0(1), values in an array are being replaced at their index instead of a new array being made
 def remove_duplicates(list)
-  raise NotImplementedError, "Not implemented yet"
+  if list.length > 1 # if list length is 1 or less, there are obviously no duplicates
+    (list.length).times do |i|
+      if list[i] == list[i+1]
+        list[i] = nil
+      end
+    end
+  end
+  return list
 end
 
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: 0(n^2), the time complexity comes from the first times loop, which loops over the `strings` array as well as each string in the array based on their respective lengths.
+# Space Complexity: 0(n), because the length of the `matches` array depends on the length of the `strings` array.
 def longest_prefix(strings)
-  raise NotImplementedError, "Not implemented yet"
+  
+  # `matches` holds the length of each matched prefix in a `strings` array
+  matches = []
+  
+  (strings.length).times do |i|
```

Comment:
It's unclear to me if this set of loops is entirely necessary.  It seems like it is doing more work than you need.